Add an issue template, tighten PR template

janetkuo opened the pull request:
Supersedes #28488

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.kubernetes.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30167)

<!-- Reviewable:end -->

thockin committed:
Add an issue template, tighten PR template

I looked at a few other examples of templates and tried to make ours more
succint and helpful.  Feedback is welcome.

Kashomon commented on .github/ISSUE_TEMPLATE.md:
Drive-by nit: s/kubctl/kubectl

Kashomon commented on .github/ISSUE_TEMPLATE.md:
This analytics tracking pixel thing seems misplaced, no? Should it exist in the issue template?

janetkuo committed:
Update PR & issue template (addressed comments, more interactive, shorter)

janetkuo commented:
<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->

**Is this a request for help** (If your answer is yes, you should instead post it in StackOverflow, Slack channel, or mailing list, see http://kubernetes.io/docs/troubleshooting/.):


**What keywords did you search in Kubernetes issues** (If you find any duplicates, you should instead reply there or use the :+1: icon in the upper-right of the issue message):

---

**Is this a BUG REPORT or FEATURE REQUEST** (choose one):

<!--
If this is a BUG REPORT, please:
  - Fill in as much of the template below as you can.  If you leave out
    information, we can't help you as well.

If this is a FEATURE REQUEST, please:
  - Describe *in detail* the feature/behavior/change you'd like to see.

In both cases, be ready for followup questions, and please respond in a timely
manner.  If we can't reproduce a bug or think a feature already exists, we
might close your issue.  If we're wrong, PLEASE feel free to reopen it and
explain why.
-->

**Kubernetes version** (use `kubectl version`):


**Environment**:
- **Cloud provider**:
- **OS**:
- **kernel**:
- **Install tools**:
- **Others**:


**What happened**:


**What you expected to happen**:


**How to reproduce it, as minimally and precisely as possible**:


**Anything else do we need to know**:



[![Analytics](https://kubernetes-site.appspot.com/UA-36037335-10/GitHub/.github/ISSUE_TEMPLATE.md?pixel)]()

janetkuo commented:
The above is the current issue template

janetkuo commented on .github/ISSUE_TEMPLATE.md:
fixed

janetkuo commented on .github/ISSUE_TEMPLATE.md:
https://github.com/kubernetes/kubernetes/pull/28488#discussion_r70337901 explains why it's here, but the problem is solved now, removed analytics

janetkuo commented:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, read our contributor guidelines https://github.com/kubernetes/kubernetes/blob/master/CONTRIBUTING.md

2. If you want faster PR reviews, read how: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/faster_reviews.md

3. Follow the instructions for writing a release note: https://github.com/kubernetes/kubernetes/blob/master/docs/devel/pull-requests.md#release-notes
-->

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, #<issue_number>, ...)` format, will close that issue when PR merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
* Enter your extended release note here (newlines are formatted as bullets).  If this PR does not warrant a release
  note, just say `NONE`.
-OR-
* Use the release-note-* labels to set the release note state (if you have access)
* Clear this block to use the PR title as the release note
```

janetkuo commented:
The above is the current issue & PR template, major changes include:

- Addressed all comments in previous PR #28488
- Make it shorter (otherwise no one is going to read it)
- Removed `PLEASE REMOVE THIS COMMENT BLOCK BEFORE SUBMITTING` (it won't display anyways so better save this space)
- Made all "read this and that paragraph" more Q&A-ish (so people can't ignore it)
- Issue template: removed developer guide https://github.com/kubernetes/kubernetes/blob/master/docs/devel/development.md, I don't think people can send a PR without first reading it, and the space in the template is expensive (this information is fighting with other more important information), so removed it

janetkuo commented:
@bgrant0607 ptal

philips commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,47 @@
+<!-- Thanks for filing an issue! Before hitting the button, please answer these questions.-->
+
+**Is this a request for help** (If yes, you should instead post it in StackOverflow, Slack channel, or mailing list, see http://kubernetes.io/docs/troubleshooting/.):
```
Perhaps something like: **Is this a request for help** (If yes, you should use our troubleshooting guide and community support channels, see http://kubernetes.io/docs/troubleshooting/.)

philips commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,47 @@
+**Is this a request for help** (If yes, you should instead post it in StackOverflow, Slack channel, or mailing list, see http://kubernetes.io/docs/troubleshooting/.):
+
+
+**What keywords did you search in Kubernetes issues before filing this one** (If you found any duplicates, you should instead reply there or use the :+1: icon in the upper-right of the issue message):
```
Missing question mark after "one?".

philips commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,47 @@
+
+**Environment**:
+- **Cloud provider**:
+- **OS**:
```
OS (e.g. from /etc/os-release)

philips commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,47 @@
+
+
+**Environment**:
+- **Cloud provider**:
```
Cloud provider or hardware configuration

philips commented on .github/ISSUE_TEMPLATE.md:
```diff
@@ -0,0 +1,47 @@
+**Environment**:
+- **Cloud provider**:
+- **OS**:
+- **kernel**:
```
Kernel (e.g. uname -a)

philips commented:
Overall this LGTM.

bgrant0607 commented on .github/PULL_REQUEST_TEMPLATE.md:
````diff
@@ -1,19 +1,23 @@
+**Special notes for your reviewer**:
+
+**Release note**:
 ```release-note
````
It would be nice if we could make the syntax clear enough to automatically set the release-note labels based on the text here.

janetkuo commented:
All comments addressed, @philips @bgrant0607 ptal

philips commented:
LGTM

janetkuo commented:
Squashed; applying tag
